sourcing-new/src/main/scala/ch/epfl/bluebrain/nexus/sourcingnew/processor/ProcessorCommand.scala

```diff
@@ -0,0 +1,112 @@
+package ch.epfl.bluebrain.nexus.sourcingnew.processor
+
+import akka.actor.typed.ActorRef
+import akka.routing.ConsistentHashingRouter.ConsistentHashable
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Command used in [[EventSourceProcessor]]
+  */
+sealed trait ProcessorCommand extends Product with Serializable
```

Comment:
I would recommend placing all subtypes in the companion object for better modularity and tc derivation.